postpic/particles/particles.py

```diff
@@ -1279,3 +1279,52 @@ def collect(self, *scalarfs):
                 particlelist[i].append(scalars[:, k])
         ret = [np.asarray(p).T for p in particlelist]
         return ret
+
+    def exportToHdf5(self, filename, *scalarfs):
+        '''
+        Exports the collected data to an hdf5-file. If the file with the same name
+        already exists, it will be deleted!
+        Parameters:
+        -----------
+        filename:   whole name or path of the file which the data will be exported to.
+        *scalarfs:  the scalarfunction(s) defining the particle property.
+                Id and weight will always be stored, so they do not have to be
+                specified here.
+
+        Returns:
+        -----------
+        The same as collect(): The list of particles with the defined properties.
+        The hdf5 file will have the structure:
+        - dset: Ids
+        - particles/Id/track: track data (ID, weight, scalarfs)
+        - particles/Id/weight: weights
+        '''
+        scalarfs = ("id", "weight") + scalarfs
+        print("The following scalars will be exported to {}: {}.".format(filename, scalarfs))
+        part_data = self.collect(*scalarfs)
+        print("Finished collecting electron data.")
+
+        timesteps_parts = []
+        for i in range(0, len(part_data)):
+            timesteps_parts.append(len(part_data[i][0]))
+
+        import h5py
+        import os
+
+        try:
+            os.remove(filename)
```

Comment:
Do you really need to remove the file first? How does h5py handle it if the file exists? Maybe its overwritten or appended? Can we have h5py hanlde this, then we do not need to change the behaviour at all. Alternatively we enforce the same behavior as `numpy.save`.